### PR TITLE
=htc FrameOutHandler and FrameEventRenderer as graph stages (#341)

### PR DESCRIPTION
**Pull request by gosubpl**
This is code twice lost that is converting FrameOutHandler and FrameEventRenderer to GraphStage(s).

### CODE REVIEW

**Comment by gosubpl** [comment]
Refs #341

**Comment by gosubpl** [comment]
Fails in the same place as https://github.com/akka/akka-http/pull/330 ... interesting. Investigating.

**Comment by gosubpl** [comment]
PLS BUILD

File: akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala [hunk 1]
```diff
@@ -318,7 +318,7 @@ class ResponseParserSpec extends FreeSpec with Matchers with BeforeAndAfterAll {
         }.concatSubstreams
 
     def collectBlocking[T](source: Source[T, Any]): Seq[T] =
-      Await.result(source.limit(100000).runWith(Sink.seq), 500.millis)
+      Await.result(source.limit(100000).runWith(Sink.seq), 1000.millis)
```
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala]: good, 500.millis is way too little.

File: akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameEventRenderer.scala [hunk 2]
```diff
@@ -14,46 +15,60 @@ import scala.annotation.tailrec
  *
  * INTERNAL API
  */
-private[http] class FrameEventRenderer extends StatefulStage[FrameEvent, ByteString] {
-  def initial: State = Idle
+private[http] final class FrameEventRenderer extends GraphStage[FlowShape[FrameEvent, ByteString]] {
+  val in = Inlet[FrameEvent]("in")
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameEventRenderer.scala]: Should be `FrameEventRenderer.in`, for better tracking what actually failed

**Comment by jrudolph** [comment]
Yep, thx @gosubpl .

**Comment by gosubpl** [comment]
@ktoso @jrudolph many thanks for support!

**Pull request merged**